shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1929,6 +1979,21 @@ - (void)continuousScrollEvent:(UIPanGestureRecognizer*)recognizer API_AVAILABLE(
       break;
     case UIGestureRecognizerStateEnded:
     case UIGestureRecognizerStateCancelled:
+      _scrollInertiaEventStartline =
+          [[NSProcessInfo processInfo] systemUptime] +
+          0.1;  // Time to lift fingers off trackpad (experimentally determined)
+      // When running an iOS app on an Apple Silicon Mac, AppKit will send an event
+      // of type UIEventTypeScroll when trackpad scroll momentum has ended. This event
+      // is sent whether the momentum ended normally or was cancelled by a trackpad touch.
+      // Since Flutter scrolling inertia will likely not match the system inertia, we should
+      // only send a PointerScrollInertiaCancel event for user-initiated cancellations.
+      // The following (curve-fitted) calculation provides a cutoff point after which any
+      // UIEventTypeScroll event will likely be from the system instead of the user.
+      _scrollInertiaEventAppKitDeadline =
+          [[NSProcessInfo processInfo] systemUptime] +
+          (0.1725 * log(hypotf([recognizer velocityInView:self.view].x,
+                               [recognizer velocityInView:self.view].y))) -
+          0.1196;
```

Comment:
Where is this formula and the magic numbers coming from? Can you add link to it?

Reply:
I collected some data myself on behavior while `runningOnMac`. Not sure where to put it, maybe in this PR comments? And I will link to this PR?

Reply:
I was wondering where did the formula `a * log (hypot(x, y)) + b` come from. 

The `log` part is particularly interesting, because scrolling inertia is typically achieved by a constant acceleration in the opposite direction, in which case the time should be linear to delta of speed (something like `dT = (v1 - v2) / a` where `dT` is delta time, `v1` is the current speed (aka `hypotf(x, y)`), `v2` is the future speed (zero), `a` is acceleration constant. I am not sure why there is a `log` or `exponential` in the formula. (high school was ages ago so i can be terribly wrong here in physics). 

Another question is how were you able to get such precise magic numbers like `0.1725` and `0.1196`.

Reply:
It is not exact, if you see my [other PR](https://github.com/flutter/flutter/pull/108298), where I attempt to match the scrolling on Mac, the stopping time cannot be computed analytically very easily. I had to use numeric method instead. This is a close approximation line of fit computed in Excel.

Reply:
Very interesting to see the non-linear change of velocity. Yes please add link to that PR with the graph, otherwise it's hard to tell where that come from.